mdf.md

```diff
@@ -0,0 +1,83 @@
+
+1. [Abstract](#abstract)
+2. [Motivation](#motivation)
+3. [Format](#format)
+    1. [Fields](#fields)
+4. [Usage](#usage)
+    1. [Informational Fields](#informational-fields)
+    2. [Configurational Fields](#configurational-fields)
+
+## Abstract
+
+In this specification, we describe a method to construct message history that will aid the consistency of [MVDS](./mvds.md), which currently do not exist, the additional fields allow for various consistency models such. Additionally we explain how data sync can be used for more lightweight messages that do not require full synchronization.
+
+## Motivation
+
+In order for more efficient synchronization of conversational messages, information should be provided allowing a node to more effectively synchronize the dependencies for any given message. Encoding a DAG within a message allows for an entire portion to be synchronized if messages are missing with a single `REQUEST` message.
+
+## Format
+
+We introduce the metadata message which is used to convey information about a message and how it SHOULD be handled. If an MVDS node implements this, we say that node has MDF capability (This might become part of a future reconciled MVDS spec).
+
+
+```protobuf
+package vac.mvds;
+
+message Metadata {
+  bytes parent = 7000;
+  bool ack_required = 7001 [default = true];
+}
+```
+
+We MAY transmit a `Metadata` message by extending the MVDS [message](./mvds.md#payloads) with a `metadata` field.
+
+```diff
+message Message {
++ Metadata metadata = 6000;
+  bytes group_id = 6001;
+  int64 timestamp = 6002;
+  bytes body = 6003;
+}
+```
+### Fields
+
+| Name                   |   Description                                                                                                                    |
+| ---------------------- | -------------------------------------------------------------------------------------------------------------------------------- |
+| `parents`               |   contains a list of parent [`message indentifier`](./mvds.md#payloads) for the specific message. |            
+| `ack_required`         |   contains a flag whether a message needs to be acknowledged or not.                                                             |
+
+## Usage
+
+The flags provided through the `Metadata` message are either informational or behavioral. Informational fields MAY be used by a node, configurational fields SHOULD be recognized and the nodes behavior SHOULD change accordingly.
+
+### Informational Fields
+
+Below are the list of informational flags and what they MAY be used for by a node.
+
+#### `parent`
+
+This field contains a list of parent [`message indentifier`](./mvds.md#payloads) for the specific message. It MUST NOT contain any messages as parent whose `ack` flag was set to `false`. This creates a Directed Acyclic Graph (DAG)<sup>1</sup> of persistent messages.
+
+This field helps establish consistency that may either be causal or eventual depending on the client.
```

Comment:
requires more explanation

also what do we mean by eventual exactly, IME it is a sloppy term

Reply:
I think eventual consistency is a well defined consistency model (what it implies is clear to me, at least the major points that apply here), what makes you think otherwise?
We can possibly explain how it applies further to the data structure being updated (dag, master-master replication), but the gist is the same as any other system: we allow inconsistent read/writes from clients, we strive to reach consistency (travelling through the dependency graph in this case), and we solve conflicts once they arise (trivial in this case as it's an append-only structure).

Reply:
https://jepsen.io/consistency doesn't even mention it explicit, as another example

Reply:
Maybe it is fine if we:
- clearly write these terms out, i.e. as full phrases with some reference
- elaborate on it being append-only (conflicts etc)
- how it relates to the DAG and upper layer expectations

Reply:
https://gun.eco/docs/Conflict-Resolution-with-Guns here's a good (albeit a bit too tutorialy) example of explaining implications and using proper nouns, describing conflicts resolution etc

Reply:
@oskarth I agree with @cammellos on this.

Reply:
I don't think it's a hair splitting contest, as long as we qualify what we mean it's fine. Right now that isn't clear, see above for ideas on how to resolve this.